.changeset/curly-bikes-train.md

```diff
@@ -0,0 +1,24 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+"__section": feature
+---
+Adds enablable allowed types to SchemaFactoryAlpha
+
+This adds the `enablable` API to [`SchemaFactoryAlpha`](https://fluidframework.com/docs/api/fluid-framework/schemafactoryalpha-class).
+Enablables can be used for schema evolution to add members to an [`AllowedTypes`](https://fluidframework.com/docs/api/fluid-framework/allowedtypes-typealias) while supporting cross version collaboration.
+
+Enablables are allowed types that can be read by the tree but not written. Attempting to write an enablable type will cause a runtime error.
+
+To add a new member to an `AllowedTypes`, add the type wrapped by `enablable`:
+```typescript
+schemaFactoryAlpha.arrayAlpha("TestArray", [schemaFactoryAlpha.number, schemaFactoryAlpha.enablable(schemaFactoryAlpha.string)]);
+```
+
+Once enough clients have this code update, it is safe to allow writing strings to the array.
+To enable writing strings to the array, a code change must be made to remove the enablable annotation:
+```typescript
+schemaFactoryAlpha.arrayAlpha("TestArray", [schemaFactoryAlpha.number, schemaFactoryAlpha.string]);
```

Comment:
I like the inclusion of an example. I think we should also have an integration test showing this feature end to end in actual code (something like what's in src/test/feature-libraries/modular-schema/schemaEvolutionExamples.spec.ts except targeting the public API, and less horrible).

Making an executable example as a test, then including the same example in the changeset or doc comments is a great pattern to help ensure everything works as promised.

I'd like to see a test with some end to end proof involving a schema upgrade showing the whole process working. If thats too big for the changeset, it can have a simplified version, but link to the full exampe code via a github link.